tests/test-snapd-desktop-integration.c

```diff
@@ -131,20 +132,19 @@ static void handle_snapd_themes_request(SoupMessage *message) {
   }
 }
 
-static void handle_snapd_get_changes_request(SoupMessage *message) {
+static void handle_snapd_get_changes_request(SoupServerMessage *message) {
   send_snapd_response(message, 200,
                       "{\"type\":\"sync\",\"status-code\":200,\"status\":"
                       "\"OK\",\"result\":{\"id\":\"1234\",\"ready\":true}}");
 }
 
-static void handle_snapd_request(SoupServer *server, SoupMessage *message,
+static void handle_snapd_request(SoupServer *server, SoupServerMessage *message,
                                  const char *path, GHashTable *query,
-                                 SoupClientContext *client,
                                  gpointer user_data) {
   if (strcmp(path, "/v2/accessories/themes") == 0) {
     handle_snapd_themes_request(message);
   } else if (strcmp(path, "/v2/changes/1234") == 0 &&
```

Comment:
What do you think about replacing all these strcmp with g_str_equal()?

Reply:
The problem with g_str_equal is it's not type safe (it takes gconstpointer as args). So it's not good for a general string comparison.